### PR TITLE
RHMAP-4395 undeploy command now in fh2 and fh3

### PR DESCRIPTION
**Pull request by pb82**
There are now two different version of the undeploy command:

- `lib/cmd/fh2/undeploy.js` works only with the old Feedhenry v2 Platform
- `lib/cmd/fh3/app/undeploy.js` is the new one that works with Feedhenry v3

The command syntax is:

`fhc app undeploy --id=<app-id> --env=<environment> --domain=<domain>`

ping @wtrocki @wei-lee

### CODE REVIEW

**Comment by fh-qube** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Reformat the code to have only one statement per line.](https://github.com/feedhenry/fh-fhc/blob/bc7ae9ebe795f26bf9907c14d25be00e4a60a9da/lib/cmd/fh2/undeploy.js#L36) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AOneStatementPerLine)

**Comment by fh-automatron** [comment]
### Pull Request Quality Guidelines

As we continue to focus on quality, here are a few things to keep in mind:

* A component owner should review your change (we've tagged these above)
* This change should be deployed to a grid somewhere (via `fhcap-cli`) - please provide a link to this
* Automated testing has been added or updated for this change (e.g. UART, Unit/Acceptance tests)
* Ensure JIRA is up to date (e.g. deployment notes, tester notes, fix version, release notes)
* There should be an associated fhcap PR with this change - please provide a link to this
* Any SonarQube issues should be addressed where applicable (use your own judgement)

Please ping a component owner (see `fh contrib`) when your builds are complete, changes are deployed and you're ready for a review.

File: lib/cmd/fh3/app/undeploy.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    desc: 'Undeploy <app-id> from environment <environment> in domain <domain>'
+  }],
+  'demand': ['id', 'env', 'domain'],
+  'alias': {
+    'id': 'i',
+    'env': 'e',
+    'domain': 'd',
+    0: 'id',
+    1: 'env',
+    2: 'domain'
```
Comment (wei-lee) [inline lib/cmd/fh3/app/undeploy.js]: does domain require required? I think it is possible to get it from the current fhc target?

**Comment by wei-lee** [comment]
Verified that this is working as expected. Although it will better if we don't ask user to provide the domain.

File: lib/cmd/fh3/app/undeploy.js (hunk 1 above)
Comment (wtrocki) [inline lib/cmd/fh3/app/undeploy.js]: Example: https://github.com/wtrocki/fh-fhc/blob/ac284808e69daaf3ccd1b0bbaad147404916ce0e/lib/cmd/fh3/app/stage.js#L7-L7

**Comment by wtrocki** [comment]
LGTM

**Comment by vchepeli** [comment]
LGTM too and agree that domain parameter is non needed here

**Comment by pb82** [comment]
Domain is now optional (if not provided, then it's read from fhc ini properties). Also added some tests for the new command.

**Pull request merged**